R/template.R

```diff
@@ -1,110 +1,184 @@
+#' @inheritParams rlang::args_dots_empty
+#'
+#' @param estimator This can either be `NULL` for the default auto-selection of
+#' averaging (`"binary"` or `"macro"`), or a single character to pass along to
+#' the metric implementation describing the kind of averaging to use.
 #'
 #' @param na_rm A `logical` value indicating whether `NA` values should be
 #' stripped before the computation proceeds. The removal is executed in
 #' `metric_vec_template()`.
 #'
-#' @param event_level For numeric metrics, this is left as `NULL` to prevent
-#' it from being passed on to the metric function implementation. For
-#' classification metrics, this can either be `NULL` to use the default
-#' `event_level` value of the `metric_fn` or a single string of either
-#' `"first"` or `"second"` to pass along describing which level should be
-#' considered the "event".
+#' @param event_level This can either be `NULL` to use the default `event_level`
+#' value of the `fn` or a single string of either `"first"` or `"second"`
+#' to pass along describing which level should be considered the `"event"`.
 #'
 #' @param case_weights For metrics supporting case weights, an unquoted
 #' column name corresponding to case weights can be passed here. If not `NULL`,
-#' the case weights will be passed on to `metric_fn` as the named argument
+#' the case weights will be passed on to `fn` as the named argument
 #' `case_weights`.
 #'
-#' @param ... Currently not used. Metric specific options are passed in
-#' through `metric_fn_options`.
 #'
-#' @param metric_fn_options A named list of metric specific options. These
+#' @param fn_options A named list of metric specific options. These
 #' are spliced into the metric function call using `!!!` from `rlang`. The
 #' default results in nothing being spliced into the call.
 #'
-#' @seealso [metric_vec_template()] [finalize_estimator()] [dots_to_estimate()]
-#'
+#' @keywords internal
 #' @export
-metric_summarizer <- function(metric_nm,
-                              metric_fn,
-                              data,
-                              truth,
-                              estimate,
-                              estimator = NULL,
-                              na_rm = TRUE,
-                              event_level = NULL,
-                              case_weights = NULL,
-                              ...,
-                              metric_fn_options = list()) {
+class_metric_summarizer <- function(name,
+                                    fn,
+                                    data,
+                                    truth,
+                                    estimate,
+                                    ...,
+                                    estimator = NULL,
+                                    na_rm = TRUE,
+                                    event_level = NULL,
+                                    case_weights = NULL,
+                                    fn_options = list()) {
+  rlang::check_dots_empty()
```

Comment:
```suggestion
  check_dots_empty()
```

We `@import rlang`. I see `rlang::` in a few other places too